.circleci/config.yml

```diff
@@ -0,0 +1,130 @@
+# Use the latest 2.1 version of CircleCI pipeline process engine.
+# See: https://circleci.com/docs/2.0/configuration-reference
+version: 2.1
+
+# Define a job to be invoked later in a workflow.
+# See: https://circleci.com/docs/2.0/configuration-reference/#jobs
+jobs:
+  build-linux:
+    # Specify the execution environment. You can specify an image from Dockerhub or use one of our Convenience Images from CircleCI's Developer Hub.
+    # See: https://circleci.com/docs/2.0/configuration-reference/#docker-machine-macos-windows-executor
+    machine:
+      image: ubuntu-2004:current
+    resource_class: large
+    # Add steps to the job
+    # See: https://circleci.com/docs/2.0/configuration-reference/#steps
+    steps:
+      - checkout
+      - run:
+          name: "Set up build tools"
+          command: |
+            sudo apt-get update 
+            sudo apt-get install -y make clang llvm lld
+            sudo apt-get install -y \
+                  libgstreamer1.0-dev \
+                  libgstreamer-plugins-base1.0-dev \
+                  libgstreamer-plugins-good1.0-dev \
+                  libglib2.0-dev \
+                  libgl1-mesa-dev \
+                  libglu1-mesa-dev \
+                  libsm-dev \
+                  libx11-dev \
+                  libx11-xcb-dev \
+                  libexpat-dev \
+                  libxkbcommon-dev \
+                  libxcb1-dev \
+                  libxcb-glx0-dev \
+                  libxcb-icccm4-dev \
+                  libxcb-image0-dev \
+                  libxcb-keysyms1-dev \
+                  libxcb-randr0-dev \
+                  libxcb-render0-dev \
+                  libxcb-render-util0-dev \
+                  libxcb-shape0-dev \
+                  libxcb-shm0-dev \
+                  libxcb-sync-dev \
+                  libxcb-xfixes0-dev \
+                  libxcb-xinerama0-dev \
+                  libxcb-xkb-dev \
+                  libxcb-util-dev
+      - run:
+          name: "Build dependencies"
+          command: make -j 4 Dependencies MODE=release
```

Comment:
Please don't use `-j 4` parameter.
It already uses all available cores for each of the project builds (but builds them sequentially).

For Dependencies it will have no effect at all (there should be a warning I think that multiple -j are used).
For our main projects, I think, you are creating a race-conditions (since Studio will not wait for Engine/QtBase libs to be created) - hence also the Makefile modification here